src/core/ddsi/src/ddsi_raweth.c

```diff
@@ -119,15 +187,8 @@ static ssize_t ddsi_raweth_conn_read (struct ddsi_tran_conn * conn, unsigned cha
       break;
     }
 
-    // FIXME: ((vtag & 0xf000) << 16)) looks decidedly odd, << 4 would make more sense
-    port = (uint32_t)(ntohs (src.sll_protocol) + ((vtag & 0xfff) << 20) + ((vtag & 0xf000) << 16));
     if (srcloc)
-    {
-      srcloc->kind = DDSI_LOCATOR_KIND_RAWETH;
-      srcloc->port = port;
-      memset(srcloc->address, 0, 10);
-      memcpy(srcloc->address + 10, src.sll_addr, 6);
-    }
+      set_locator(srcloc, src.sll_addr, ntohs (src.sll_protocol), vtag);
```

Comment:
Is it really the case that `vtag` a.k.a. `pauxd->tp_vlan_tci` is already in native byte order? I'm asking only because it seems somewhat against the pattern, but it is not impossible.

Reply:
I this case the vlan tag is already in host order